### PR TITLE
Update Kotlin to 1.7.20-RC

### PR DESCRIPTION
**Pull request by IgnatBeresnev**

### CODE REVIEW

File: runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/kotlin/kotlinCompilationUtils.kt [hunk 1]
```diff
@@ -10,26 +10,36 @@ import org.jetbrains.kotlin.gradle.plugin.KotlinSourceSet
...
 }
 
-internal fun Project.compilationsOf(sourceSet: KotlinSourceSet): List<KotlinCompilation> {
-    //KT-45412 Make sure .kotlinSourceSets and .allKotlinSourceSets include the default source set
-    return allCompilationsOf(sourceSet).filter { compilation -> sourceSet in compilation.kotlinSourceSets || sourceSet == compilation.defaultSourceSet }
-}
-
-private fun KotlinMultiplatformExtension.allCompilationsOf(sourceSet: KotlinSourceSet): List<KotlinCompilation> {
-    val allCompilations = targets.flatMap { target -> target.compilations }
-    return allCompilations.filter { compilation -> sourceSet in compilation.allKotlinSourceSets || sourceSet == compilation.defaultSourceSet }
```
Comment (IgnatBeresnev) [inline runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/kotlin/kotlinCompilationUtils.kt:30]: It failed with `Overload resolution ambiguity`, so refactored it a bit

```
e: /home/ignat/IdeaProjects/dokka/runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/kotlin/kotlinCompilationUtils.kt: (17, 51): Overload resolution ambiguity: 
private fun [Error type: Type for error type constructor (KotlinSingleTargetExtension)].allCompilationsOf(sourceSet: KotlinSourceSet): List<org.jetbrains.dokka.gradle.kotlin.KotlinCompilation /* = org.jetbrains.kotlin.gradle.plugin.KotlinCompilation<KotlinCommonOptions> */> defined in org.jetbrains.dokka.gradle.kotlin in file kotlinCompilationUtils.kt
private fun KotlinMultiplatformExtension.allCompilationsOf(sourceSet: KotlinSourceSet): List<org.jetbrains.dokka.gradle.kotlin.KotlinCompilation /* = org.jetbrains.kotlin.gradle.plugin.KotlinCompilation<KotlinCommonOptions> */> defined in org.jetbrains.dokka.gradle.kotlin in file kotlinCompilationUtils.kt
e: /home/ignat/IdeaProjects/dokka/runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/kotlin/kotlinCompilationUtils.kt: (18, 12): One type argument expected. Use 'KotlinSingleTargetExtension<*>' if you don't want to pass type arguments
e: /home/ignat/IdeaProjects/dokka/runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/kotlin/kotlinCompilationUtils.kt: (25, 12): Overload resolution ambiguity: 
private fun [Error type: Type for error type constructor (KotlinSingleTargetExtension)].allCompilationsOf(sourceSet: KotlinSourceSet): List<org.jetbrains.dokka.gradle.kotlin.KotlinCompilation /* = org.jetbrains.kotlin.gradle.plugin.KotlinCompilation<KotlinCommonOptions> */> defined in org.jetbrains.dokka.gradle.kotlin in file kotlinCompilationUtils.kt
internal fun Project.allCompilationsOf(sourceSet: KotlinSourceSet): List<org.jetbrains.dokka.gradle.kotlin.KotlinCompilation /* = org.jetbrains.kotlin.gradle.plugin.KotlinCompilation<KotlinCommonOptions> */> defined in org.jetbrains.dokka.gradle.kotlin in file kotlinCompilationUtils.kt
e: /home/ignat/IdeaProjects/dokka/runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/kotlin/kotlinCompilationUtils.kt: (25, 50): Cannot infer a type for this parameter. Please specify it explicitly.
e: /home/ignat/IdeaProjects/dokka/runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/kotlin/kotlinCompilationUtils.kt: (33, 13): One type argument expected for class KotlinSingleTargetExtension<TARGET : KotlinTarget>
e: /home/ignat/IdeaProjects/dokka/runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/kotlin/kotlinCompilationUtils.kt: (34, 41): Cannot infer a type for this parameter. Please specify it explicitly
```

File: kotlin-analysis/src/main/kotlin/org/jetbrains/dokka/analysis/AnalysisEnvironment.kt [hunk 2]
```diff
@@ -362,9 +362,11 @@ class AnalysisEnvironment(val messageCollector: MessageCollector, val analysisPl
...
             ): ResolverForModule =
                 CommonResolverForModuleFactory(
-                    CommonAnalysisParameters { content ->
-                        environment.createPackagePartProvider(content.moduleContentScope)
-                    },
+                    CommonAnalysisParameters(
+                        metadataPartProviderFactory = { content ->
+                            environment.createPackagePartProvider(content.moduleContentScope)
+                        }
+                    ),
```
Comment (IgnatBeresnev) [inline kotlin-analysis/src/main/kotlin/org/jetbrains/dokka/analysis/AnalysisEnvironment.kt:369]: A new primary constructor arg has been added (doesn't look like we need it), so have to use named parameters

File: runners/gradle-plugin/src/test/kotlin/org/jetbrains/dokka/gradle/GradleDokkaSourceSetBuilderTest.kt [hunk 3]
```diff
@@ -113,7 +113,7 @@ class GradleDokkaSourceSetBuilderTest {
         sourceSet.dependsOn(sourceSet.DokkaSourceSetID("s3"))
         sourceSet.dependsOn(GradleDokkaSourceSetBuilder("s4", project))
         sourceSet.dependsOn(GradleDokkaSourceSetBuilder("s5", project).build())
-        sourceSet.dependsOn(DefaultKotlinSourceSet(project, "s6"))
```
Comment (IgnatBeresnev) [inline runners/gradle-plugin/src/test/kotlin/org/jetbrains/dokka/gradle/GradleDokkaSourceSetBuilderTest.kt:116]: Looks like there was some refactoring of gradle api, `DefaultKotlinSourceSet` is now abstract, so cannot instantiate it directly.

**Review by vmishenev — COMMENTED** [review]
Can you update integration tests?

**Comment by IgnatBeresnev** [comment]
> Can you update integration tests?

Oh yeah, indeed. I had a feeling I forgot to do something, that was it :)

**Comment by IgnatBeresnev** [comment]
JFYI serialization and coroutine integration tests cannot be run with Kotlin 1.7.20 yet - there's native/npm compilation failures. So I didn't update them

**Review by vmishenev — APPROVED** [review]

**Pull request merged**